app/modules/scan/service.server.ts

```diff
@@ -138,3 +145,61 @@ export async function getScanByQrId({ qrId }: { qrId: string }) {
     });
   }
 }
+
+export async function createScanNote({ userId, qrId, latitude, longitude, manuallyGenerated }: {
+  userId?: string | null;
+  qrId: string;
+  latitude?: Scan["latitude"];
+  longitude?: Scan["longitude"];
+  manuallyGenerated?: boolean;
+}) {
+  try {
+    let message = "";
+    const { assetId, organizationId } = await getQr(qrId)
+    if (assetId) {
+      if (userId && userId != "anonymous") {
+        const user = await db.user.findFirst({
+          where: {
+            id: userId,
+          },
+          select: {
+            firstName: true,
+            lastName: true,
+          },
+        });
+
+        const userName = user?.firstName + " " + user?.lastName
+        if (manuallyGenerated) {
+          message = `**${userName}** manually updated the GPS coordinates to **${latitude} ${longitude}**`
+        } else {
+          message = `**${userName}** performed a scan of the asset QR code`
+        }
+        return await createNote({
+          content: message,
+          type: "UPDATE",
+          userId,
+          assetId
+        });
+      } else {
+        if (organizationId) {
```

Comment:
This should not happen, that there is no organizationId. If there is an assetId there will always be organization id. PLease annotate that we are just doing this extra check but it should never happen.